build.gradle

```diff
@@ -289,6 +291,11 @@ subprojects {
       exceptionFormat = testExceptionFormat
     }
     logTestStdout.rehydrate(delegate, owner, this)()
+
+    retry {
+      maxRetries = 3
+      maxFailures = 50
```

Comment:
I would go with something conservative: 1 retry and max 5 failures.We want to fix flaky tests, but if something just fails once, it's not worth failing the build given Jenkins flakiness.

Reply:
We can adjust after we see how it works in practice, but this would be a good starting point IMO.

Reply:
Ok. I figured we can also make these configurable. Updated the PR.